src/net/p2p.rs

```diff
@@ -3004,14 +3030,16 @@ impl PeerNetwork {
                 PeerNetworkWorkState::Prune => {
                     // did one pass
                     did_cycle = true;
-
-                    // clear out neighbor connections after we finish sending
-                    if self.do_prune {
-                        do_prune = true;
-
-                        // re-enable neighbor walks
-                        self.do_prune = false;
-                    }
+                    do_prune = true;
+                    /*
+                    do_prune = match self.walk {
+                        Some(ref mut walk) => {
+                            walk.do_prune = false;
+                            true
+                        }
+                        None => true
+                    };
+                    */
```

Comment:
Should we add a TODO or remove that snippet?

Reply:
The snippet should be removed; will do.